lib/resources/FileUploads.js

```diff
@@ -9,14 +10,45 @@ module.exports = StripeResource.extend({
 
   overrideHost: 'uploads.stripe.com',
 
-  requestDataProcessor: function(method, data, headers) {
+  requestDataProcessor: function(method, data) {
```

Comment:
This got a bit more complicated than I expected  - I imagined something very similar to what you'd done previously, except that the entire `requestDataProcessor` function would just get an additional `callback` parameter that was always used (either sync or async), and the main code flow would either hit some 'default' function with callback, or the one specified here.

Reply:
@jlomas-stripe Definitely a bit more complicated but I believe I vetted the scenario you've mentioned and the nature of `streams` posed a problem. If I'm understanding your comment (correct me if I'm wrong), the gist is: `requestDataProcessor` would send the buffered `multipart/form-data` back to `_request` via the additional `callback` - which ever way handled (ie. data processed via `stream` or already buffered data). From there `_request` can process as normal.

The trouble I encountered is that when the `stream` is set by attaching an event listener `.on('data')`, the code is handled via `process.nextTick()` which runs the callback after the rest of the code and before the event loop. If I remember correctly, this caused `data` to be passed back to the `callback` `undefined` and then proceed through `_request` without the buffered data. The way I figured, returns a function in both scenarios so we can wait for the `stream` to finish processing. This will guarantee the buffered data exists when the rest of `_request` is processed by invoking the `setHeadersAndMakeRequest` within the `callback` from either scenario. My apologies if that was long-winded! :slightly_smiling_face:

If that logic may be flawed or there's a better way to go about this - I'm fine with scrapping this solution for the sake of clarity and going back to the drawing board. Let me know!

Reply:
@jlomas-stripe ohhh no I misunderstood! All good, yea that'll definitely work inside `_request` - I have some free time later and can put it to action. thanks for further explaining!

Reply:
@jlomas-stripe just tested it and worked perfectly. I appreciate the help simplifying. I'll update the PR with the changes requested in the review.